### PR TITLE
History Management and Ui fixes

### PR DESCRIPTION
**Pull request by surbhi84**
This branch consists of the below features

- The videos that are opened goes to the history.
- You can remove one history at a time with the button provided.
- Changed the sidebar design.
- Minimized the login signup popups.

ps- to create history you need to login.

### CODE REVIEW

**Comment by govardhanreddyc79** [comment]
Clean and easily readable code.
Good Folder structure.
Great work

**Review by anupkgurung — APPROVED** [review]
clean and structured code.
nice breakdown of components

File: src/backend/controllers/HistoryController.js [hunk 1]
```diff
@@ -54,14 +54,16 @@ export const addVideoToHistoryHandler = function (schema, request) {
...
-        }
-      );
+      return new Response(201, {}, { history: user.history });
+      // return new Response(
+      //   409,
+      //   {},
+      //   {
+      //     errors: ["The video is already in your history"],
+      //   }
+      // );
```
Comment (Aditi0505) [inline src/backend/controllers/HistoryController.js:66]: Please remove if not needed.

File: src/components/Navbar.jsx [hunk 2]
```diff
@@ -1,31 +1,44 @@
...
   async function logoutHandler() {
     try {
-      // const response = await authLogin(email, password);
       userDispatch({ type: LOGOUT });
       navigate("/");
-      // setTriggerLogin(false);
+      setTriggerLogin(false);
       setSuccessToast({ show: true, msg: "You are Logged out!" });
     } catch (err) {
       console.error(err);
```
Comment (Aditi0505) [inline src/components/Navbar.jsx:23]: Display the error on UI

File: src/components/Navbar.jsx [hunk 3]
```diff
@@ -1,31 +1,44 @@
...
     }
   }
+
+  useEffect(() => {
+    if (triggerLogin) document.body.style.overflow = "hidden";
+    else {
+      document.body.style.overflow = "auto";
```
Comment (Aditi0505) [inline src/components/Navbar.jsx:30]: Just wondering if we can do it using classes? What is the need to access document.body?

File: src/hooks/reducer/userReducer/reducer.js [hunk 4]
```diff
@@ -26,6 +34,22 @@ export const useUserReducer = () => {
...
+          user: { ...state.user, history: [...state.user.history, payload] },
+        };
+      }
+      case DELHISTORY: {
+        const newHistory = state.user.history.filter((i) => i.id !== payload);
+        return { ...state, user: { ...state.user, history: newHistory } };
+      }
+      // case GETHISTORY: {
+      //   return { ...state, user: { ...state.user, history : {...state.user.history} } };
+      // }
```
Comment (Aditi0505) [inline src/hooks/reducer/userReducer/reducer.js:49]: Please remove if not needed

**Review by Aditi0505 — APPROVED** [review]
Left few suggestion. Rest looks good to merge.

File: src/components/Navbar.jsx (hunk 3 above)
Comment (surbhi84) [reply src/components/Navbar.jsx:30]: I was just trying something out, forgot to remove it... i'll remove it thanks.

**Pull request merged**